### ISSUE TITLE
Use osm-changeset-classification for flagging bad changesets?

### ISSUE THREAD

**Issue by jremillard** [mapbox/osmcha-frontend#289]
I'm submitting a feature request

My deep learning changeset classifier at https://github.com/jremillard/osm-changeset-classification is close to working. Are you interested in using it? It can detect imports, SPAM, bad tagging.

**Comment by willemarcel** [comment]
Hi @jremillard 

Congratulations for the project! It's always good to see OSM QA projects.

Are the results of your changeset classifier available in some place? I would need to see how effective it is before discussing with the team about it.

**Comment by jremillard** [comment]
I am still working on the project. Currently I am getting 95% validation accuracy, using this much data.

Import - 871 changesets
OK - 8201 changesets
SPAM - 322 changesets
Tagging Error - 213 changesets

The changests are categorized in the trainingdata\changesets.csv file. If you have a set of validation changesets numbers for the underlying detector in osmcha, I would love to have them. Honestly, it would be a big if the only thing we did together was share our human classified changeset list.

I will touch base in a couple of weeks when I have the training data all cleaned up.

**Comment by batpad** [comment]
@jremillard this looks great! I know @bkowshik was working on https://github.com/mapbox/gabbar which attempted to use classification from osmcha to train a model on various attributes. Hopefully you two can connect on both datasets used and approaches.

I like your approach of trying to stick to 2 fairly simple to understand classifications of "Spam" and "Imports" and the results look great!

You can definitely use the OSMCha API to query for changesets that have been marked "harmful" by humans, and more recently, with some "Tags" applied. You can find the API docs here: https://osmcha.mapbox.com/api-docs/

Feel free to join the OSMCha mailing list: https://lists.openstreetmap.org/listinfo/osmcha-dev to ask about API related questions if they seem unrelated to this ticket :-)

**Comment by batpad** [comment]
For the initial question of using the classifier on OSMCha - again, I'd suggest doing the classification on your end and then use the OSMCha API to post "suspicious reasons" to OSMCha - this seems like interesting work and I'd be curious / interested to see something like this running on an on-going basis and see what the results look like (and feeding potential human review on osmcha back into the model training).

Hoping you and @willemarcel can work out adding these results with some custom "reasons" to OSMCha :)
